tests/test_services.py

```
@@ -83,39 +91,168 @@ def test_optimization_service_handles_solver_failure():
 
 def test_optimization_service_catches_runtime_exceptions():
-    """Verify that the service captures and reports unexpected runtime errors."""
+    """Verify that the service captures and re-raises unexpected runtime errors."""
     df = pd.DataFrame({"Name": ["P1"], "S1": [10.0]})
-    target = "src.core.services.solver_interface.run_optimization"
-    with patch(target, side_effect=RuntimeError("Fail")):
-        _, metrics = OptimizationService.run(
-            df,
+    # Cause an error inside the try block
+    with patch("src.core.services.Participant", side_effect=RuntimeError("Fail")):
+        with pytest.raises(RuntimeError, match="Fail"):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)